### DISCUSSION TITLE
Ability to skip converting movie "sample files" for Torrents

### DISCUSSION THREAD

**Discussion by Dzhuneyt** [ptr727/PlexCleaner#341] [Ideas]
Usually movies distributed via Torrent trackers have a so-called "sample file" which is a truncated version of the whole movie, as a way to download it first and get a feel for the quality of the release for that movie.

Converting those with PlexCleaner is a bit of an overkill. I know these are pretty small, in comparison to the whole movie, but for high quality releases (e.g. 4K) and weak transcoding hardware (e.g. Synology NAS' CPU), they can eat up a lot of TIME as an absolute value. Relatively, compared to the overall optimization time, it's something like 2%, but if a 2TB collection takes 5 days to optimize fully on this weak hardware, 2% of 5 days is still 2.5 hours wasted.

I was wondering if we can conditionally skip optimization of such Sample files, given that they follow a few predictable naming patterns, e.g. being nested in a subfolder called /sample, being named `sample.mkv`, etc.

**Comment by ptr727** [comment]
Would some kind of skip filenames config, like the extensions skip, work for this?
Is wildcard or regex support required, or just a list of filenames?

**Reply by Dzhuneyt** [reply]
Some torrent clients append something like `.partial` to the filename, as it is being downloaded, and finally rename it to the final filename.

But not all of them do. Some of them just start a file from 0 bytes and keep appending the downloaded data.

That's why I thought exposing some sort of parameter in PlexCleaner to ignore files created sooner than X hours, would help to solve all of this. This would allow the user to pick something reasonable, after which any movie is definitely fully downloaded, and ready for optimization by PlexCleaner.

**Reply by ptr727** [reply]
Filename wildcard skipping is something I am looking into, but file date/time created/modified is in my experience too unreliable, and if you do want to use that maybe use a script the periodically runs, does the timestamp checking, and then calls the process command.

As for writing in-place, the tooling does test for open for write, and tooling that are writing should only share for read while writing, but again, better to use other (xrr) tools that rename/move the files to the desired destination when done, and then monitor can process them.

**Reply by ptr727** [reply]
Give the `develop` build a try, it will upgrade your settings on first run, then you can add wildcards, e.g.

```jsonc
  "ProcessOptions": {
    // Files to skip, use * and ? wildcards, non-ascii characters must be JSON escaped
    "FileIgnoreMasks": [
      "*.ass",
      "*.fuse_hidden*",
      "*.jpg",
      "*.nfo",
      "*.partial~",
      "*.sample.*",
      "*.sample",
      "*.smi",
      "*.srt",
      "*.ssa",
      "*.vtt"
    ],
```
